Dev Mode fails when using Variables and Includes in Server.xml

### Summary
When using the latest version of the OpenLiberty Maven Plugin (3.2.4) on a clean system (empty `.m2/` directory) I am seeing `Caused by: java.lang.IllegalArgumentException: Illegal group reference` as part of the `mvn liberty:dev` build when there is a variable and include used in the `server.xml`.

### Details
The IBM Stock Trader sample's portfolio microservice makes use of a [variable](https://github.com/IBMStockTrader/portfolio/blob/master/src/main/liberty/config/server.xml#L30) and an [include stanza](https://github.com/IBMStockTrader/portfolio/blob/master/src/main/liberty/config/server.xml#L44) to define the security configuration the server should use. 

If you comment out the include statement, the `mvn liberty:dev` completes successfully and the server starts in development mode. 

Likewise, If you run this configuration outside of maven using the standard OpenLiberty `bin/server start` (with the include statement in place) the server starts successfully and uses the default variable value of `basic` with the include to properly configure basic security for the server. 

### Recreate
Steps to recreate:

1. Clear out your local Maven repo cache.  On my Mac I run `rm -rf ~/.m2/repository/*`
1. Clone the repo https://github.com/IBMStockTrader/portfolio
1. Update the pom.xml in two places:
    1. Update the `<parent>` to use version 3.2.4 of the `liberty-maven-app-parent`
    1. Change in the `<plugins>` stanza to read:
`  <groupId>net.wasdev.wlp.maven.plugins</groupId>
                        <artifactId>liberty-maven-plugin</artifactId>
                        <version>2.4.2</version>` 
to be
`<groupId>io.openliberty.tools</groupId>
                        <artifactId>liberty-maven-plugin</artifactId>
                        <version>3.2.4</version>`
1. run `mvn -e liberty:dev`. You will eventually see:
```
ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.2.4:dev (default-cli) on project portfolio: Illegal group reference -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.2.4:dev (default-cli) on project portfolio: Illegal group reference
...
Caused by: java.lang.IllegalArgumentException: Illegal group reference
    at java.util.regex.Matcher.appendExpandedReplacement (Matcher.java:1068)
    at java.util.regex.Matcher.appendReplacement (Matcher.java:908)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.evaluateExpression (ServerFeatureUtil.java:467)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.parseIncludeNode (ServerFeatureUtil.java:345)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.getServerXmlFeatures (ServerFeatureUtil.java:286)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.getServerFeatures (ServerFeatureUtil.java:119)
    at io.openliberty.tools.maven.InstallFeatureSupport.getInstalledFeatures (InstallFeatureSupport.java:168)
    at io.openliberty.tools.maven.server.InstallFeatureMojo.installFeatures (InstallFeatureMojo.java:67)
    at io.openliberty.tools.maven.server.InstallFeatureMojo.doExecute (InstallFeatureMojo.java:50)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:119)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runLibertyMojo (StartDebugMojoSupport.java:227)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runLibertyMojoInstallFeature (StartDebugMojoSupport.java:221)
    at io.openliberty.tools.maven.server.DevMojo.doExecute (DevMojo.java:705)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

Sure. Gave 3.3 a try. 

First, [the parent is referencing a 3.3-M5 milestone](https://repo1.maven.org/maven2/io/openliberty/tools/liberty-maven-app-parent/3.3/liberty-maven-app-parent-3.3.pom) that doesn't exist so I had to leave it at 3.2.4:
```
    <parent>
        <groupId>io.openliberty.tools</groupId>
        <artifactId>liberty-maven</artifactId>
        <version>3.3</version>
    </parent>
    
    <artifactId>liberty-maven-app-parent</artifactId>
    <packaging>pom</packaging>
    
    <name>Liberty Maven Plugin Parent</name>
    <description>Liberty Maven Plugin Parent</description>
    <url>https://github.com/OpenLiberty/ci.maven</url>
    
    <licenses>
        <license>
            <name>The Apache Software License, Version 2.0</name>
            <url>https://raw.github.com/OpenLiberty/ci.maven/master/LICENSE</url>
            <distribution>repo</distribution>
        </license>
    </licenses>
    
    <build>
        <pluginManagement>
            <plugins>
                <plugin>
                    <groupId>io.openliberty.tools</groupId>
                    <artifactId>liberty-maven-plugin</artifactId>
                    <version>3.3-M5-SNAPSHOT</version>
```

Leaving the parent at 3.2.4 and updating the plugin to 3.3 I experience the same problem:
```
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.2.4:dev (default-cli) on project portfolio: Illegal group reference -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.2.4:dev (default-cli) on project portfolio: Illegal group reference
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
```

Hi @cherylking, I see the 3.3.1 version of the plugins are available in Maven Central so I updated my pom.xml to pull in the latest release.  Same issues as before:
```
[ERROR] Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.3.1:dev (default-cli) on project portfolio: Illegal group reference -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.openliberty.tools:liberty-maven-plugin:3.3.1:dev (default-cli) on project portfolio: Illegal group reference
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Illegal group reference
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:129)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Illegal group reference
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:137)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:119)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runLibertyMojo (StartDebugMojoSupport.java:260)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runLibertyMojoInstallFeature (StartDebugMojoSupport.java:253)
    at io.openliberty.tools.maven.server.DevMojo.runLibertyMojoInstallFeature (DevMojo.java:1174)
    at io.openliberty.tools.maven.server.DevMojo.doExecute (DevMojo.java:783)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.IllegalArgumentException: Illegal group reference
    at java.util.regex.Matcher.appendExpandedReplacement (Matcher.java:1068)
    at java.util.regex.Matcher.appendReplacement (Matcher.java:908)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.evaluateExpression (ServerFeatureUtil.java:467)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.parseIncludeNode (ServerFeatureUtil.java:345)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.getServerXmlFeatures (ServerFeatureUtil.java:286)
    at io.openliberty.tools.common.plugins.util.ServerFeatureUtil.getServerFeatures (ServerFeatureUtil.java:119)
    at io.openliberty.tools.maven.InstallFeatureSupport.getInstalledFeatures (InstallFeatureSupport.java:168)
    at io.openliberty.tools.maven.server.InstallFeatureMojo.installFeatures (InstallFeatureMojo.java:67)
    at io.openliberty.tools.maven.server.InstallFeatureMojo.doExecute (InstallFeatureMojo.java:50)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.twdata.maven.mojoexecutor.MojoExecutor.executeMojo (MojoExecutor.java:119)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runLibertyMojo (StartDebugMojoSupport.java:260)
    at io.openliberty.tools.maven.server.StartDebugMojoSupport.runLibertyMojoInstallFeature (StartDebugMojoSupport.java:253)
    at io.openliberty.tools.maven.server.DevMojo.runLibertyMojoInstallFeature (DevMojo.java:1174)
    at io.openliberty.tools.maven.server.DevMojo.doExecute (DevMojo.java:783)
    at org.codehaus.mojo.pluginsupport.MojoSupport.execute (MojoSupport.java:122)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -43,8 +43,7 @@ jobs:
     - name: Checkout ci.common
       uses: actions/checkout@v3
       with:
-        repository: cherylking/ci.common
-        ref: includeLocationVarResolution
+        repository: OpenLiberty/ci.common
         path: ci.common
     - name: Checkout ci.ant
       uses: actions/checkout@v3
@@ -99,7 +98,7 @@ jobs:
     - name: Clone ci.ant, ci.common, ci.maven repos to C drive
       run: |
         cp -r D:/a/ci.maven/ci.maven C:/ci.maven
-        git clone https://github.com/cherylking/ci.common.git --branch includeLocationVarResolution --single-branch C:/ci.common
+        git clone https://github.com/OpenLiberty/ci.common.git C:/ci.common
         git clone https://github.com/OpenLiberty/ci.ant.git C:/ci.ant
     - name: Cache maven packages
       uses: actions/cache@v3

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -22,13 +22,13 @@ jobs:
       matrix:
         # test against latest update of each major Java version, as well as specific updates of LTS versions:
         RUNTIME: [ol, wlp]
-        RUNTIME_VERSION: [22.0.0.12, 22.0.0.9]
+        RUNTIME_VERSION: [22.0.0.12, 23.0.0.3]
         java: [17, 11, 8]
         exclude:
         - java: 8
-          RUNTIME_VERSION: 22.0.0.9
+          RUNTIME_VERSION: 22.0.0.12
         - java: 17
-          RUNTIME_VERSION: 22.0.0.9
+          RUNTIME_VERSION: 22.0.0.12
     name: ${{ matrix.RUNTIME }} ${{ matrix.RUNTIME_VERSION }}, Java ${{ matrix.java }}, Linux
     steps:
     # Checkout repos
@@ -43,7 +43,8 @@ jobs:
     - name: Checkout ci.common
       uses: actions/checkout@v3
       with:
-        repository: OpenLiberty/ci.common
+        repository: cherylking/ci.common
+        ref: includeLocationVarResolution
         path: ci.common
     - name: Checkout ci.ant
       uses: actions/checkout@v3
@@ -76,13 +77,13 @@ jobs:
       matrix:
         # test against latest update of each major Java version, as well as specific updates of LTS versions:
         RUNTIME: [ol, wlp]
-        RUNTIME_VERSION: [22.0.0.12, 22.0.0.9]
+        RUNTIME_VERSION: [22.0.0.12, 23.0.0.3]
         java: [17, 11, 8]
         exclude:
         - java: 8
-          RUNTIME_VERSION: 22.0.0.9
+          RUNTIME_VERSION: 22.0.0.12
         - java: 17
-          RUNTIME_VERSION: 22.0.0.9
+          RUNTIME_VERSION: 22.0.0.12
     name: ${{ matrix.RUNTIME }} ${{ matrix.RUNTIME_VERSION }}, Java ${{ matrix.java }}, Windows
     steps:    
     # Checkout repos
@@ -98,7 +99,7 @@ jobs:
     - name: Clone ci.ant, ci.common, ci.maven repos to C drive
       run: |
         cp -r D:/a/ci.maven/ci.maven C:/ci.maven
-        git clone https://github.com/OpenLiberty/ci.common.git C:/ci.common
+        git clone https://github.com/cherylking/ci.common.git --branch includeLocationVarResolution --single-branch C:/ci.common
         git clone https://github.com/OpenLiberty/ci.ant.git C:/ci.ant
     - name: Cache maven packages
       uses: actions/cache@v3

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/ServerFeatureSupport.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/ServerFeatureSupport.java
@@ -20,6 +20,7 @@ import java.io.IOException;
 import java.util.ArrayList;
 import java.util.LinkedHashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 
 import org.apache.maven.execution.ProjectDependencyGraph;
@@ -99,9 +100,10 @@ public class ServerFeatureSupport extends BasicSupport {
      * @param suppressLogs if true info and warning will be logged as debug
      * @return instance of ServerFeatureUtil
      */
-    protected ServerFeatureUtil getServerFeatureUtil(boolean suppressLogs) {
+    protected ServerFeatureUtil getServerFeatureUtil(boolean suppressLogs, Map<String, File> libDirPropFiles) {
         if (servUtil == null) {
             createNewServerFeatureUtil();
+            servUtil.setLibertyDirectoryPropertyFiles(libDirPropFiles);
         }
         if (suppressLogs) {
             servUtil.setSuppressLogs(true);

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java
@@ -296,9 +296,9 @@ public class DevMojo extends LooseAppSupport {
                     keepTempDockerfile, mavenCacheLocation, upstreamProjects, recompileDeps, project.getPackaging(),
                     pom, parentPoms, generateFeatures, compileArtifactPaths, testArtifactPaths, webResourceDirs);
 
-            ServerFeatureUtil servUtil = getServerFeatureUtil(true);
             this.libertyDirPropertyFiles = BasicSupport.getLibertyDirectoryPropertyFiles(installDir, userDir,
-                    serverDirectory);
+                    serverDirectory);                    
+            ServerFeatureUtil servUtil = getServerFeatureUtil(true, libertyDirPropertyFiles);           
             this.existingFeatures = servUtil.getServerFeatures(serverDirectory, libertyDirPropertyFiles);
             this.upstreamMavenProjects = upstreamMavenProjects;
         }
@@ -1011,7 +1011,7 @@ public class DevMojo extends LooseAppSupport {
         @Override
         public void installFeatures(File configFile, File serverDir, boolean generateFeatures) {
             try {
-                ServerFeatureUtil servUtil = getServerFeatureUtil(true);
+                ServerFeatureUtil servUtil = getServerFeatureUtil(true, libertyDirPropertyFiles);
                 Set<String> features = servUtil.getServerFeatures(serverDir, libertyDirPropertyFiles);
                 if (features != null) {
                     Set<String> featuresCopy = new HashSet<String>(features);
@@ -1048,7 +1048,7 @@ public class DevMojo extends LooseAppSupport {
         @Override
         public ServerFeatureUtil getServerFeatureUtilObj() {
             // suppress logs from ServerFeatureUtil so that dev console is not flooded
-            return getServerFeatureUtil(true);
+            return getServerFeatureUtil(true, libertyDirPropertyFiles);
         }
 
         @Override
@@ -1058,7 +1058,7 @@ public class DevMojo extends LooseAppSupport {
 
         @Override
         public void updateExistingFeatures() {
-            ServerFeatureUtil servUtil = getServerFeatureUtil(true);
+            ServerFeatureUtil servUtil = getServerFeatureUtil(true, libertyDirPropertyFiles);
             Set<String> features = servUtil.getServerFeatures(serverDirectory, libertyDirPropertyFiles);
             existingFeatures = features;
         }

--- a/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java
+++ b/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/GenerateFeaturesMojo.java
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright IBM Corporation 2021, 2022.
+ * (C) Copyright IBM Corporation 2021, 2023.
  *
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not
  * use this file except in compliance with the License. You may obtain a copy of
@@ -170,7 +170,7 @@ public class GenerateFeaturesMojo extends ServerFeatureSupport {
         // Set<String> featuresToInstall = getSpecifiedFeatures(null); 
 
         // get existing server features from source directory
-        ServerFeatureUtil servUtil = getServerFeatureUtil(true);
+        ServerFeatureUtil servUtil = getServerFeatureUtil(true, null);
 
         Set<String> generatedFiles = new HashSet<String>();
         generatedFiles.add(GENERATED_FEATURES_FILE_NAME);

